prog/weavewait/main.go

```diff
@@ -39,3 +63,79 @@ func checkErr(err error) {
 		os.Exit(1)
 	}
 }
+
+// Prepend the addresses to /etc/hosts, removing the docker address
+func updateHosts(iface *net.Interface) {
+	addrs, err := iface.Addrs()
+	checkErr(err)
+	if len(addrs) == 0 {
+		return
+	}
+	hostname, err := os.Hostname()
+	checkErr(err)
+
+	hosts := parseHosts()
+
+	// Remove existing ips pointing to our hostname
+	toRemove := []string{}
+	for ip, addrs := range hosts {
+		for _, addr := range addrs {
+			if addr == hostname {
+				toRemove = append(toRemove, ip)
+				break
+			}
+		}
+	}
+	for _, ip := range toRemove {
+		delete(hosts, ip)
+	}
+
+	// Add the weave ip(s)
+	for _, addr := range addrs {
+		if addr, ok := addr.(*net.IPNet); ok {
+			ip := addr.IP.String()
+			hosts[ip] = append(hosts[ip], hostname)
+		}
+	}
+
+	writeHosts(hosts)
+}
+
+func parseHosts() map[string][]string {
```

Comment:
This returns a map of ip -> []hostname, right?

I wonder if you made this hostname -> []ip, the code above might be simpler?